src/App.tsx

```diff
@@ -1,19 +1,74 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import { getTodos } from './api/todos';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotifications } from './components/ErrorNotifications';
+import { Filters } from './components/Filters';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoList } from './components/TodoList';
+
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.ALL);
+  const [hasError, setHasError] = useState(false);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const loadTodos = async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch {
+        setHasError(true);
+      }
+    }
+  };
+
+  const handleFilter = (filterType: FilterType) => {
+    setFilterBy(filterType);
+  };
+
+  const handleCloseError = () => {
+    setHasError(false);
+  };
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    loadTodos();
   }, []);
 
+  const filteredTodos = todos.filter(todo => {
+    switch (filterBy) {
+      case FilterType.ALL:
+        return todo;
+
+      case FilterType.ACTIVE:
+        return !todo.completed;
+
+      case FilterType.COMPLETED:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
```

Comment:
It's better to keep visibleTodos in the state then a separate variable in the component